src/core/gui/PageView.cpp

```diff
@@ -664,7 +659,16 @@ auto XojPageView::onButtonReleaseEvent(const PositionInputData& pos) -> bool {
         ToolHandler* h = control->getToolHandler();
         bool isDrawingTypeSpline = h->getDrawingType() == DRAWING_TYPE_SPLINE;
         if (!isDrawingTypeSpline || !this->inputHandler->getStroke()) {  // The Spline Tool finalizes drawing manually
-            assert(hasNoViewOf(overlayViews, inputHandler));
+            if (!hasNoViewOf(overlayViews, inputHandler)) {
+                /*
+                 * This is a workaround to avoid mem leaks and segfaults
+                 * This `if` statement can probably be replaced by an `assert` once #4377 is fixed
+                 */
+                g_warning("InputHandler still has a View before deletion in XojPageView::onButtonReleaseEvent. "
+                          "Deleting it");
+                std::remove_if(overlayViews.begin(), overlayViews.end(),
+                               [ih = this->inputHandler](auto& v) { return v->isViewOf(ih); });
```

Comment:
Actually, just to be safe, we need to erase all possible matches:
```suggestion
                overlayViews.erase(std::remove_if(overlayViews.begin(), overlayViews.end(),
                               [ih = this->inputHandler](auto& v) { return v->isViewOf(ih); }), overlayViews.end());
```
I created a static function `eraseViewsOf` doing just that, to avoid making the same mistake again and again.